### PR TITLE
Add support for Python 3.10 and 3.11, drop EOL 3.6

### PR DESCRIPTION
**Pull request by hugovk**
Test 3.10 and 3.11, and add Trove classifiers.

Drop 3.6, EOL since 2021-09-03.

Also bump various versions:

* GitHub Actions
* docs dependencies
* dev dependencies

### CODE REVIEW

File: sphinxext/opengraph/__init__.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def get_tags(
         parse_result = urlparse(config["html_baseurl"])
 
         if config["html_baseurl"] is None:
-            raise EnvironmentError("ReadTheDocs did not provide a valid canonical URL!")
```
Comment (TheTripleV) [inline sphinxext/opengraph/__init__.py:75]: Why was this changed?

Comment (hugovk) [reply sphinxext/opengraph/__init__.py:75]: I ran https://github.com/asottile/pyupgrade/ to upgrade to modern syntax:

According to the [Python docs](https://docs.python.org/3/library/exceptions.html?highlight=environmenterror#EnvironmentError), `EnvironmentError` is kept for compatibility with previous versions; starting from Python 3.3, it's an alias of `OSError`. So we can use `OSError` instead.

**Comment by TheTripleV** [comment]
Can this be closed in favor of #70 then?

**Comment by hugovk** [comment]
Yep, closing in favour of #70.
